lib/lightning_web/channels/run_channel.ex

```diff
@@ -176,10 +175,10 @@ defmodule LightningWeb.RunChannel do
   def handle_in("step:complete", payload, socket) do
     %{
       "run_id" => socket.assigns.run.id,
-      "project_id" => socket.assigns.project_id
+      "project_id" => socket.assigns.project_id,
+      "wipe?" => socket.assigns.retention_policy == :erase_all
     }
     |> Enum.into(payload)
-    |> maybe_drop_dataclip(socket.assigns.retention_policy)
     |> Runs.complete_step()
```

Comment:
Instead of extending the payload that goes into `complete_step`, I think we should have a `complete_step(map(), rentention_policy() \\ :retain_all)` (I'm assuming what the options are for policies here). 

_or_ we get the projects retention policy from here?

Basically we have a function inside `Runs` that is there to handle these kinds of decisions...

Not sure what we can/should do right now - but I don't love extending string maps that change behaviour based on information we have control of (i.e. project settings vs params sent from the worker)